### PR TITLE
Fixed external idp iframe issue.

### PR DESCRIPTION
**Pull request by Meruja**
Fixed wso2/product-apim/issues/7850
Depends on : wso2/product-apim/pull/7951

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 1]
```diff
@@ -2196,6 +2196,44 @@ private static boolean isWSDL2Resource(byte[] wsdl) throws APIManagementExceptio
...
+            return idpEndpoint;
+        }
+    }
+
+    /**
+     * Get the check session URL to load in the session management iframe
+     *
+     * @return URL to be used in iframe source for the check session with IDP
+     */
+
```
Comment (rmsamitha) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:2223]: Remove empty line

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 2]
```diff
@@ -2196,6 +2196,44 @@ private static boolean isWSDL2Resource(byte[] wsdl) throws APIManagementExceptio
...
+    public static String getExternalIDPCheckSessionEndpoint() throws APIManagementException {
+
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        String oidcCheckSessionEndpoint = config.getFirstProperty(
+                APIConstants.IDENTITY_PROVIDER_OIDC_CHECK_SESSION_ENDPOINT);
+        if (oidcCheckSessionEndpoint == null) {
+            return getServerURL() + "/oidc/checksession";
```
Comment (Sarangan0219) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:2231]: Can we use a constant for "/oidc/checksession"

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 1 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:2223]: Fixed with 6578fa6

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 2 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:2231]: Fixed with 6578fa6

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/utils.js [hunk 3]
```diff
@@ -30,3 +30,11 @@ var getLoopbackOrigin = function() {
     var origin = 'https://' + app.origin.host + ":" + mgtTransportPort;
     return origin; // Unless there is a port offset this is https://localhost:9443
 };
+
+function getIDPOrigin(){
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/utils.js:34]: Formatting issue

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/utils.js [hunk 4]
```diff
@@ -30,3 +30,11 @@ var getLoopbackOrigin = function() {
...
+    return utils.getExternalIDPOrigin();
+}
+
+function getIDPCheckSessionEndpoint(){
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/utils.js:38]: Formatting issue

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 5]
```diff
@@ -28,6 +28,14 @@ var getLoopbackOrigin = function() {
     return origin;
 };
 
+function getIDPOrigin(){
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js:31]: Formatting issue

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 6]
```diff
@@ -28,6 +28,14 @@ var getLoopbackOrigin = function() {
...
+    return utils.getExternalIDPOrigin();
+}
+
+function getIDPCheckSessionEndpoint(){
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js:35]: Formatting issue

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/utils.js (hunk 3 above)
Comment (Meruja) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/utils.js:34]: Fixed with https://github.com/wso2/carbon-apimgt/pull/8939/commits/a8a6b4129f323381aa0bb47d61f3098bc310ca70

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/utils.js (hunk 4 above)
Comment (Meruja) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/utils.js:38]: Fixed with https://github.com/wso2/carbon-apimgt/pull/8939/commits/a8a6b4129f323381aa0bb47d61f3098bc310ca70

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js (hunk 5 above)
Comment (Meruja) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js:31]: Fixed with https://github.com/wso2/carbon-apimgt/pull/8939/commits/a8a6b4129f323381aa0bb47d61f3098bc310ca70

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js (hunk 6 above)
Comment (Meruja) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js:35]: Fixed with https://github.com/wso2/carbon-apimgt/pull/8939/commits/a8a6b4129f323381aa0bb47d61f3098bc310ca70

**Pull request merged**